pkg/lib/oauth/oidc/id_token_test.go

```diff
@@ -206,3 +206,174 @@ func TestIDTokenIssuer(t *testing.T) {
 		)
 	})
 }
+
+func TestIDTokenIssuer_GetUserInfo(t *testing.T) {
+	Convey("GetUserInfo", t, func() {
+		ctrl := gomock.NewController(t)
+
+		now := time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC)
+		createdAt := now.Add(-1 * time.Hour)
+		updatedAt := now.Add(-30 * time.Minute)
+
+		mockUserInfoService := NewMockUserInfoService(ctrl)
+		mockUserInfoService.EXPECT().GetUserInfoBearer(gomock.Any(), "user-id").Return(
+			&userinfo.UserInfo{
+				User: &model.User{
+					IsAnonymous:       false,
+					IsVerified:        true,
+					CanReauthenticate: true,
+				},
+				EffectiveRoleKeys: []string{"role-1", "role-3"},
+				Authenticators: []model.UserInfoAuthenticator{
+					{
+						CreatedAt: createdAt,
+						UpdatedAt: updatedAt,
+						Type:      model.AuthenticatorTypePassword,
+						Kind:      model.AuthenticatorKindPrimary,
+					},
+					{
+						CreatedAt: createdAt,
+						UpdatedAt: updatedAt,
+						Type:      model.AuthenticatorTypeOOBSMS,
+						Kind:      model.AuthenticatorKindPrimary,
+						Phone:     "+85298765432",
+					},
+					{
+						CreatedAt: createdAt,
+						UpdatedAt: updatedAt,
+						Type:      model.AuthenticatorTypeOOBEmail,
+						Kind:      model.AuthenticatorKindPrimary,
+						Email:     "test@example.com",
+					},
+					{
+						CreatedAt:   createdAt,
+						UpdatedAt:   updatedAt,
+						Type:        model.AuthenticatorTypeTOTP,
+						Kind:        model.AuthenticatorKindPrimary,
+						DisplayName: "Google Authenticator",
+					},
+				},
+			},
+			nil,
+		)
+
+		issuer := &IDTokenIssuer{
+			UserInfoService: mockUserInfoService,
+		}
+
+		clientConfig := &config.OAuthClientConfig{
+			ClientID:        "client-id",
+			ApplicationType: config.OAuthClientApplicationTypeSPA,
+		}
+		client := oauth.ClientClientLike(clientConfig, []string{"openid", "email", oauth.FullUserInfoScope, string(model.ClaimAuthenticators), string(model.ClaimPhoneNumber), string(model.ClaimEmail)})
+		userInfo, err := issuer.GetUserInfo(context.Background(), "user-id", client)
+		So(err, ShouldBeNil)
+		So(userInfo, ShouldResemble, map[string]interface{}{
```

Comment:
We should serialize userinfo and assert on the actual JSON. This can also test we have put correct JSON struct tags.